src/ray/core_worker/context.cc

```diff
@@ -264,8 +260,9 @@ void WorkerContext::SetCurrentActorId(const ActorID &actor_id) LOCKS_EXCLUDED(mu
 void WorkerContext::SetTaskDepth(int64_t depth) { task_depth_ = depth; }
 
 void WorkerContext::SetCurrentTask(const TaskSpecification &task_spec) {
-  GetThreadContext().SetCurrentTask(task_spec);
   absl::WriterMutexLock lock(&mutex_);
+  InitExecContext(task_spec.TaskId());
+  current_exec_context_->SetCurrentTask(task_spec);
```

Comment:
should we call `GetExecContextInternal` ?

Reply:
We could - it's kind of implicit, where after initing the context, `current_exec_context_` will also point to the current ctx (same thread) with the writer lock acquired. 
So it's still correct but I guess calling `GetExecContextInternal` will be more readable as you pointed out.